tests/test_plots.py

```diff
@@ -1,238 +1,318 @@
+        file_type, compare_output)
+
+    match = 'must be a coordinates object.'
+    with pytest.raises(ValueError, match=match):
+        function('coords')
+
+
+@pytest.mark.parametrize('function', [
+    (sp.plot.balloon),
+    (sp.plot.balloon_wireframe),
+    (sp.plot.contour),
+    (sp.plot.contour_map),
+    (sp.plot.pcolor_map),
+    (sp.plot.pcolor_sphere)])
+def test_spherical_default(function):
+    """Test all spherical plots with default arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+
+    # do plotting
+    filename = f'default_{function.__name__}'
+    create_figure()
+    function(coords, data)
+    save_and_compare(
+        create_baseline, baseline_path, output_path, filename,
+        file_type, compare_output)
+
+    # test error for invalid inputs
+    match = 'data must be a 1D array with the same cshape as the coordinates'
+    with pytest.raises(ValueError, match=match):
+        function(coords, 'data')
+
+    match = 'coordinates must be a coordinates object.'
+    with pytest.raises(ValueError, match=match):
+        function('coords', data)
+
+
+@pytest.mark.parametrize('function', [
+    (sp.plot.balloon),
+    (sp.plot.balloon_wireframe),
+    (sp.plot.contour),
+    (sp.plot.contour_map),
+    (sp.plot.pcolor_map),
+    (sp.plot.pcolor_sphere)])
+@pytest.mark.parametrize('cmap', [plt.get_cmap('plasma'), 'plasma', None])
+def test_spherical_cmap(function, cmap):
+    """Test all spherical plots with custom arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+
+    # do plotting
+    if isinstance(cmap, mpl.colors.Colormap):
+        # if a colormap object is passed, use its name for the filename
+        cmap_str = 'ColormapObject'
+    else:
+        # otherwise use the string representation of the colormap
+        cmap_str = str(cmap)
+    filename = f'cmap_{function.__name__}_{cmap_str}'
+    create_figure()
+    function(coords, data, cmap=cmap)
+    save_and_compare(create_baseline, baseline_path, output_path, filename,
+                     file_type, compare_output)
+
+    # test error for invalid inputs
+    match = 'cmap'
+    with pytest.raises(ValueError, match=match):
+        function(coords, data, cmap=5)
+
+
+@pytest.mark.parametrize('function', [
+    (sp.plot.balloon),
+    (sp.plot.balloon_wireframe),
+    (sp.plot.contour),
+    (sp.plot.contour_map),
+    (sp.plot.pcolor_map),
+    (sp.plot.pcolor_sphere)])
+@pytest.mark.parametrize('colorbar', [True, False])
+def test_spherical_colorbar(function, colorbar):
+    """Test all spherical plots with custom arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+
+    # do plotting
+    filename = f'colorbar_{function.__name__}_{colorbar}'
```

Comment:
```suggestion
    filename = f'colorbar_{function.__name__}_cb_{colorbar}'
```
Something like this makes identifying the baseline file a lot easier.